### PR TITLE
21953 Furnishings job - Implement stage one letter flow

### PR DESCRIPTION
**Pull request by leodube-aot**
*Issue #:* /bcgov/entity#21952

*Description of changes:*
- Implemented stage 1 letter flow logic (excluding letter/pdf creation and SFTP to BCMail+)
- Added unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: jobs/furnishings/src/furnishings/stage_processors/stage_one.py [hunk 1]
```diff
@@ -41,33 +44,85 @@ async def process(self, batch_processing: BatchProcessing):
...
             # send paper letter if business is still not in good standing after 5 days of email letter sent out
-            pass
+            valid_furnishing_names = [
+                Furnishing.FurnishingName.DISSOLUTION_COMMENCEMENT_NO_AR,
+                Furnishing.FurnishingName.DISSOLUTION_COMMENCEMENT_NO_TR,
+                Furnishing.FurnishingName.DISSOLUTION_COMMENCEMENT_NO_AR_XPRO,
+                Furnishing.FurnishingName.DISSOLUTION_COMMENCEMENT_NO_TR_XPRO
+            ]
+            tz = pytz.timezone('UTC')
+            today_date = tz.localize(datetime.today())
```
Comment (leodube-aot) [inline jobs/furnishings/src/furnishings/stage_processors/stage_one.py:58]: `furnishing.created_date` is a UTC timezone, so we need to compare to today in UTC.

File: jobs/furnishings/src/furnishings/stage_processors/stage_one.py [hunk 2]
```diff
@@ -41,33 +44,85 @@ async def process(self, batch_processing: BatchProcessing):
...
             # send paper letter if business doesn't have email address
-            pass
+            new_furnishing = self._create_new_furnishing(
+                batch_processing,
+                eligible_details,
+                Furnishing.FurnishingType.MAIL
+            )
+            # TODO: create and add letter to either AR or transition pdf
+            # TODO: send AR and transition pdf to BCMail+
+            new_furnishing.status = Furnishing.FurnishingStatus.PROCESSED
```
Comment (chenhongjing) [inline jobs/furnishings/src/furnishings/stage_processors/stage_one.py:106]: Think `processed_date` should be populated, and similarly around line 125.

**Comment by chenhongjing** [comment]
Wondering, is there any further updates/decisions for address logic on which address is used to send paper letter? 💭

**Comment by leodube-aot** [comment]
> Wondering, is there any further updates/decisions for address logic on which address is used to send paper letter? 💭

Thanks for reminding me, I had spoken to Argus about it and updated the address model but totally forgot to update the implementation here 😅

**Review by chenhongjing — APPROVED** [review]

File: jobs/furnishings/src/furnishings/config.py [hunk 3]
```diff
@@ -84,6 +84,8 @@ class _Config:  # pylint: disable=too-few-public-methods
     TESTING = False
     DEBUG = False
 
+    SECOND_NOTICE_DELAY = os.getenv('SECOND_NOTICE_DELAY', None)
```
Comment (argush3) [inline jobs/furnishings/src/furnishings/config.py:87]: can we do the int casting here like we do in dissolutions job?  Think it's safe to just default to 5 as well if nothing returns for whatever reason.

https://github.com/bcgov/lear/blob/2685686f797871a80267f110401f3cbfde59e911/jobs/involuntary-dissolutions/config.py#L84

File: jobs/furnishings/src/furnishings/stage_processors/stage_one.py [hunk 4]
```diff
@@ -31,6 +33,7 @@ def __init__(self, app, qsm):
         self._app = app
         self._qsm = qsm
 
+        self._second_notice_delay = int(app.config.get('SECOND_NOTICE_DELAY'))
```
Comment (argush3) [inline jobs/furnishings/src/furnishings/stage_processors/stage_one.py:36]: can remove the casting once config.py has been updated as per comment above

**Review by argush3 — APPROVED** [review]

**Pull request merged**